src/hooks/useAudioDeviceTooltipSelection.tsx

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React, { MutableRefObject, useRef, useState } from "react";
+
+import { toLeftOrRightOf } from "../components/structures/ContextMenu";
+import IconizedContextMenu, {
+    IconizedContextMenuOptionList,
+    IconizedContextMenuRadio,
+} from "../components/views/context_menus/IconizedContextMenu";
+import { _t } from "../languageHandler";
+import MediaDeviceHandler, { MediaDeviceKindEnum } from "../MediaDeviceHandler";
+import { requestMediaPermissions } from "../utils/media/requestMediaPermissions";
+
+interface State {
+    devices: MediaDeviceInfo[];
+    device: MediaDeviceInfo | null;
+    showDeviceSelect: boolean;
+}
+
+export const useAudioDeviceTooltipSelection = (
```

Comment:
In terms of responsibilities, I'm not sure we should embed the logic and the presentation in the hook.

As far as I understand we should leverage hooks to extract the logic into reusable functions. The presentation should be a component itself.
Otherwise we'll end up writing another hooks  that does all the permission requests all over again when we need it in another context